src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1280,12 +1281,11 @@ public static UriHostNameType CheckHostName(string? name)
                             return UriHostNameType.IPv6;
                         }
                     }
+                }
 
-                    end = name.Length;
-                    if (IPv4AddressHelper.IsValid(fixedName, 0, ref end, false, false, false) && end == name.Length)
-                    {
-                        return UriHostNameType.IPv4;
-                    }
+                if (IPv4AddressHelper.IsValid(name.AsSpan(), out end, false, false, false) && end == name.Length)
```

Comment:
This `AsSpan()` is necessary because `IPv4AddressHelper.IsValid` is generic. This needs to remain so - it's in the path of IPv6 address parsing (used in the case of IPv4 addresses embedded inside IPv6 addresses.)